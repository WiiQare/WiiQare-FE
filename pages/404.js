import Link from "next/link";
import React from "react";
import { BiLeftArrow } from "react-icons/bi";
import Image from "next/image";

const Error404 = () => {
  return (
    <>
      <div className="min-w-screen min-h-screen bg-blue-100 flex items-center p-5 lg:p-20 overflow-hidden relative">
        <div className="flex-1 min-h-full min-w-full rounded-3xl bg-white shadow-xl p-10 lg:p-20 text-gray-800 relative md:flex items-center text-center md:text-left">
          <div className="w-full md:w-1/2">
            <div className="mb-10 lg:mb-20">
              <Image
                alt="logo"
                src="/images/logo_dark.png"
                width={40}
                height={40}
                className="h-10"
              />
            </div>
            <div className="mb-10 md:mb-20 text-gray-600 font-light">
              <h1 className="font-black uppercase text-3xl lg:text-5xl text-orange mb-10">
                You seem to be lost!
              </h1>
              <p>The page you&apos;re looking for isn&apos;t available.</p>
              <p>Try searching again or use the Go Back button below.</p>
            </div>
            <div className="mb-20 md:mb-0">
              <Link href="/" legacyBehavior>
                <a className="text-lg font-light outline-none focus:outline-none transform transition-all hover:scale-110 text-orange hover:text-yellow-600 flex items-center gap-1">
                  <BiLeftArrow /> Go Back
                </a>
              </Link>
            </div>
          </div>
          <div className="w-full md:w-1/2 text-center">
            <Image alt="404" src="/images/404.png" height={40} width={40} />
            <a
              href="https://www.freepik.com/vectors/business"
              target="_blank"
              className="text-xs text-gray-300"
            >
              WiiQare Ã  votre service
            </a>
          </div>
        </div>
        <div className="w-64 md:w-96 h-96 md:h-full bg-blue-200 bg-opacity-30 absolute -top-64 md:-top-96 right-20 md:right-32 rounded-full pointer-events-none -rotate-45 transform"></div>
        <div className="w-96 h-full bg-yellow-200 bg-opacity-20 absolute -bottom-96 right-64 rounded-full pointer-events-none -rotate-45 transform"></div>
      </div>

      <div className="flex items-end justify-end fixed bottom-0 right-0 mb-4 mr-4 z-10">
        <div>
          <a
            title="Buy me a beer"
            href="#"
            target="_blank"
            className="bg-white flex items-center justify-center w-16 h-16 rounded-full transition-all shadow hover:shadow-lg transform hover:scale-110 hover:rotate-12"
          >
            <Image
              alt="favicon"
              className="object-cover object-center w-full h-full rounded-full"
              src="/images/favicon.png"
              width={40}
              height={40}
            />
          </a>
        </div>
      </div>
    </>
  );
};

export default Error404;
