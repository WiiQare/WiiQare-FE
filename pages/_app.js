import Head from "next/head";
import { SessionProvider } from "next-auth/react";
import { ThemeProvider } from "@mui/material/styles";
import { theme } from "../theme";

import "../styles/globals.css";
import "../styles/main.css";

export default function App({ Component, pageProps }) {
  const getLayout = Component.getLayout ?? ((page) => page);
  return (
    <>
      <Head>
        <meta
          name="description"
          content="Payé les soins à vos familles depuis l'étranger"
        />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/favicon.ico" />
        <script src="https://unpkg.com/flowbite@1.5.3/dist/flowbite.js"></script>

      </Head>
      <SessionProvider session={pageProps.session}>
        <ThemeProvider theme={theme}>        
          {getLayout(<Component {...pageProps} />)}
        </ThemeProvider>
      </SessionProvider>
    </>
  );
}
