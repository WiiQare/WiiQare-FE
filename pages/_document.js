import { Html, Head, Main, NextScript } from 'next/document';
import Drawer from '../components/molecules/Navbar/Drawer';

export default function Document() {
  return (
    <Html lang="en">
      <Head>
        <link rel="manifest" href="/manifest.json" />
        <link rel="apple-touch-icon" href="/images/favicon-192x192.png"></link>
        <meta name="theme-color" content="#fff" />
      </Head>
      <script src="https://unpkg.com/flowbite@1.5.3/dist/flowbite.js"></script>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
}
