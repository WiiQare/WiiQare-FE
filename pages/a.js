import React, { useEffect } from 'react';
import { useSession } from "next-auth/react";
import Router from "next/router";

const A = () => {
    const {status, data} = useSession();

    useEffect(() => {
       if(status === 'unauthenticated') Router.replace('/login')
    }, [status]);

    if (status === 'authenticated') 
        return (
            <div>
                Protected
            </div>
        )

    return <>Loading...</>
}

export default A;
