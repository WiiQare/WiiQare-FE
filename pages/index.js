import Head from 'next/head'
import { getSession, useSession, signOut } from "next-auth/react"

export default function Home() {

	const { data:session } = useSession()
	
	const handleSignOut = () => {
		signOut()
	}

	return (
		<>
			<Head>
				<title>JAMII - UNICEF CRYPTO</title>
				<meta name="description" content="Payé les soins à vos familles depuis l'étranger" />
				<meta name="viewport" content="width=device-width, initial-scale=1" />
				<link rel="icon" href="/favicon.ico" />
			</Head>
			

			{session ? <User session={session} handleSignOut={handleSignOut} /> : <Guest />}
		</>
	)
}

function User({session, handleSignOut}) {
	return (
		<main>
			Homepage
		</main>
	)
}

function Guest() {
	return (
		<main className='container mx-auto text-center py-20 space-y-8'>
			<h3 className="text-4xl font-bold">Guest User</h3>
		</main>
	)
}

export async function getServerSideProps({req}) {
	const session = await getSession({req})

	if(!session) {
		return {
			redirect: {
				destination: '/login',
				permanent: false
			}
		}
	}

	return {props: {session}}
}