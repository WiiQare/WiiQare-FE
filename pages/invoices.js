import Head from "next/head";
import { getSession } from "next-auth/react";
import DashboardLayout from "../layouts/Dashboard";

const Page = () => {
  return (
    <>
      <Head>
        <title>Invoices</title>
      </Head>
      <div>
        <h2 className="text-black font-w700 mt-4">Invoices</h2>
      </div>
    </>
  );
};

Page.getLayout = (page) => (
  <DashboardLayout className="space-y-8">{page}</DashboardLayout>
);

export default Page;

export async function getServerSideProps({ req }) {
  const session = await getSession({ req });

  if (!session) {
    return {
      redirect: {
        destination: "/login",
        permanent: false,
      },
    };
  }

  return { props: { session } };
}
