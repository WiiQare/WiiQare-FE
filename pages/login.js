import Logo from "../components/atoms/LogoHeader";
import WelcomeSlider from "../components/molecules/WelcomeSlider";
import LoginForm from "../components/organisms/Auth/LoginForm";
import { LOGIN_SLIDES } from "../utils/constants";
import { useMediaQuery, CssBaseline } from "@mui/material";
import Head from "next/head";
import React from "react";

function Login() {
  const matches = useMediaQuery("(max-width: 992px)");
  return (
    <>
      <Head>
        <title>Wiiqare | Login</title>
        <link
          rel="stylesheet"
          href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.2/dist/css/bootstrap.min.css"
          integrity="sha384-xOolHFLEh07PJGoPkLv1IbcEPTNtaed2xpHsD9ESMhqIYd0nLMwNLD69Npy4HI+N"
          crossOrigin="anonymous"
        />
      </Head>
      <CssBaseline />
      <div id="form-section" className="container-fluid signin">
        <Logo />
        <div className="row">
          {!matches && <WelcomeSlider slides={LOGIN_SLIDES} />}
          <LoginForm />
        </div>
      </div>
    </>
  );
}

export default Login;
