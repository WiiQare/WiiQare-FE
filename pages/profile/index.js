import Head from "next/head";
import { getSession } from "next-auth/react";
import DashboardLayout from "../../layouts/Dashboard";
import Profile from "../../components/organisms/Profile";

const Page = () => {
  return (
    <>
      <Head>
        <title>My Profile</title>
      </Head>
        <Profile />
    </>
  );
};

Page.getLayout = (page) => (
  <DashboardLayout className="space-y-8">{page}</DashboardLayout>
);

export default Page;

export async function getServerSideProps({ req }) {
  const session = await getSession({ req });

  if (session) {
    return {
      redirect: {
        destination: "/login",
        permanent: false,
      },
    };
  }

  return { props: { session } };
}
