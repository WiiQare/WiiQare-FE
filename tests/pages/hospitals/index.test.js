import Page from '@/pages/hospitals';
import { render } from '@testing-library/react';
import { DrawContext } from '@/pages/_app';
import { SessionProvider } from 'next-auth/react';
require('jest-fetch-mock').enableMocks();
import DashboardLayout from '../../../layouts/Dashboard';

fetch.mockResponse('[]');

describe('Hospitals', () => {
  it('should render the page', () => {
    const { container } = render(
      <SessionProvider session={{ user: { data: { userId: 'random123' } } }}>
        <DrawContext.Provider
          value={{
            draw: false,
            setDraw: jest.fn(),
            hideSide: false,
            setHideSide: jest.fn(),
          }}
        >
          <Page />
        </DrawContext.Provider>
      </SessionProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should use DashboardLayout as layout', () => {
    expect(Page.getLayout(<div />)).toEqual(
      <DashboardLayout className="space-y-8">
        <div />
      </DashboardLayout>,
    );
  });
});
