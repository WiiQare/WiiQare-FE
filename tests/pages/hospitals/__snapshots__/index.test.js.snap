// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hospitals should render the page 1`] = `
<div>
  <div
    class="p-2 space-y-6 md:py-8 md:px-6 mb-20 md:mb-6"
  >
    <div
      class="border w-full rounded-lg py-8 px-6 flex flex-col flex-wrap md:flex-row justify-between items-center gap-4 bg-white cursor-pointer"
    >
      <div
        class="flex flex-col"
      >
        <h1
          class="md:text-2xl text-gray-700 font-bold flex-wrap flex items-center gap-2 text-lg"
        >
          Liste des hôpitaux
        </h1>
      </div>
      <div
        class="flex flex-row flex-wrap  gap-2 center justify-end space-x-2"
      >
        <button
          class="text-white bg-blue-700 border border-transparent hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 disabled:hover:bg-blue-700 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 dark:disabled:hover:bg-blue-600 focus:!ring-2 group flex h-min items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg bg-primary"
          type="button"
        >
          <span
            class="flex items-center rounded-md text-sm px-4 py-2"
          >
            <svg
              class="mr-2 h-5 w-5"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="Save_Down_2"
              >
                <g>
                  <path
                    d="M18.437,20.948H5.563a2.372,2.372,0,0,1-2.5-2.21v-11a2.372,2.372,0,0,1,2.5-2.211h.462a.5.5,0,0,1,0,1H5.563a1.38,1.38,0,0,0-1.5,1.211v11a1.38,1.38,0,0,0,1.5,1.21H18.437a1.38,1.38,0,0,0,1.5-1.21v-11a1.38,1.38,0,0,0-1.5-1.211h-.462a.5.5,0,0,1,0-1h.462a2.372,2.372,0,0,1,2.5,2.211v11A2.372,2.372,0,0,1,18.437,20.948Z"
                  />
                  <path
                    d="M15.355,10.592l-3,3a.5.5,0,0,1-.35.15.508.508,0,0,1-.36-.15l-3-3a.5.5,0,0,1,.71-.71l2.14,2.139V3.552a.508.508,0,0,1,.5-.5.5.5,0,0,1,.5.5v8.49l2.15-2.16a.5.5,0,0,1,.71.71Z"
                  />
                </g>
              </g>
            </svg>
            Télécharger le rapport
          </span>
        </button>
        <div
          class="w-[auto] dropdown dropdown-end flex"
        >
          <label
            class=" bg-white border rounded-xl w-full py-2 px-4 flex justify-between items-center py-1 w-[auto]"
            tabindex="0"
          >
            <span
              class="flex mr-2 gap-4 items-center"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  id="Circle_List"
                >
                  <g>
                    <path
                      d="M20.438,6.062h-9a.5.5,0,0,1,0-1h9a.5.5,0,0,1,0,1Z"
                    />
                    <path
                      d="M20.438,12.5h-9a.5.5,0,0,1,0-1h9a.5.5,0,0,1,0,1Z"
                    />
                    <path
                      d="M20.438,18.935h-9a.5.5,0,1,1,0-1h9a.5.5,0,0,1,0,1Z"
                    />
                    <path
                      d="M5.562,8.062a2.5,2.5,0,1,1,2.5-2.5A2.5,2.5,0,0,1,5.562,8.062Zm0-4a1.5,1.5,0,1,0,1.5,1.5A1.5,1.5,0,0,0,5.562,4.062Z"
                    />
                    <path
                      d="M5.562,14.5a2.5,2.5,0,1,1,2.5-2.5A2.5,2.5,0,0,1,5.562,14.5Zm0-4a1.5,1.5,0,1,0,1.5,1.5A1.5,1.5,0,0,0,5.562,10.5Z"
                    />
                    <path
                      d="M5.562,20.938a2.5,2.5,0,1,1,2.5-2.5A2.5,2.5,0,0,1,5.562,20.938Zm0-4a1.5,1.5,0,1,0,1.5,1.5A1.5,1.5,0,0,0,5.562,16.938Z"
                    />
                  </g>
                </g>
              </svg>
              <span
                class="font-light"
              >
                Trier par
              </span>
            </span>
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
                fill-rule="evenodd"
              />
            </svg>
          </label>
          <ul
            class="undefined w-full mt-14 dropdown-content menu p-2 bg-white border rounded-box text-gray-700 font-light"
            tabindex="0"
          >
            <li>
              <a>
                Detail
              </a>
            </li>
            <li>
              <a>
                Annuler
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="border rounded-lg overflow-x-auto w-full relative"
    >
      Loading...
    </div>
  </div>
</div>
`;
